### PR TITLE
Windows Layout

### PR DESCRIPTION
**Pull request by korenkovv**

### CODE REVIEW

File: WebClient/src/core/window/windowManager.css [hunk 1]
```diff
@@ -0,0 +1,8 @@
+.window-manager-dock-highligh {
```
Comment (cornelissenja) [inline WebClient/src/core/window/windowManager.css]: "highligh" typo?

File: WebClient/src/core/window/windowView.js [hunk 2]
```diff
@@ -15,6 +15,10 @@ var WindowView = L.Evented.extend({
         this._maxWidth = options.maxWidth || null;
         this._maxHeight = options.maxHeight || null;
 
+        this._left = null;
```
Comment (cornelissenja) [inline WebClient/src/core/window/windowView.js]: Naming of fields is not what i expected: _left and _currentLeft do not give a hint what is stored in them.. current/persistent might be an option?

**Review by cornelissenja — CHANGES_REQUESTED** [review]
Only a few small remarks

File: WebClient/doc/core/window/window.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+*WindowView* (*window*) is a base class for all windows in the application. To show a *window* on the screen  it should be added to *WindowManager* which will take care about focusing, positioning and docking them. When docking a window *WindowManager* uses *Layout* abstraction which is injected into the *WindowManager*. *Layout* defines how docked windows should be placed and is responsible for tracking docked windows and resizing and repositioning them when needed. *Layout* is an abstraction with [defined interface](./layout/layout.md) that every concrete layout should implement.
+
+*WindowManager* tries to dock a *window* when it is added to the *WindowManager*. *Window* asks *WindowManager* to dock or undock itself when corresponding button is pressed on the *window*. *WindowManager* delegates the search of docked position to a *Layout*. *Layout* takes control over *window*'s position and size after *window* is inserted into the *Layout*.
+
+The sequence diagram below illustrates interactions between *WindowView* (*window*), *WindowManager*, *Layout* and user.
+
+![WindowArchitecture](./images/WindowArchitecture-Sequence.svg)
```
Comment (rajatSinha303) [inline WebClient/doc/core/window/window.md:17]: In the 'add window' and 'dock window' sequence diagrams, the layout is always returning true. Isn't there supposed to be a 'hasFreeSpace()' check which decides on the returned result being either true or false

File: WebClient/src/core/window/perimeterLayout.js [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+                height: cellHeight
+            };
+
+            if (cell.layoutItem) {
+                cell.layoutItem.fillSpace(cell.geometry);
+            }
+        });
+    },
+
+    _calculateCellWidht: function () {
```
Comment (rajatSinha303) [inline WebClient/src/core/window/perimeterLayout.js]: spelling mistake

**Review by rajatSinha303 — CHANGES_REQUESTED** [review]

File: WebClient/doc/core/window/window.md (hunk 3 above)
Comment (korenkovv) [reply WebClient/doc/core/window/window.md:17]: It is a sequence diagram therefore it shows only certain flows to illustrate the idea. Such diagrams are not ment to be generic but they show certain flow e.g. when `insert()` function returns `true`. Generally speaking, there should be as many sequence diagrams as sequences. In the current document I wanted to show only the happy flow to give the idea and not presenting all possible sequences.

**Pull request merged**